docs/pages/guides/upload.mdx

```diff
@@ -0,0 +1,85 @@
+import { Callout } from 'nextra/components'
+
+# Uploading Books
+
+Stump has an opt-in feature that allows your users to upload books to your server.
+
+<Callout emoji="⚠️">
+	This functionality allows files to be uploaded to your server, so it should be used with caution.
+	This feature must be manually enabled by the server owner by setting the `enable_upload`
+	configuration variable to `true`.
+</Callout>
+
+## Enabling upload
+
+Uploading is **disabled by default**, and is gated behind the `enable_upload` configuration variable (`true` enables uploading). This variable can be set by modifying your `Stump.toml` file to set `enable_upload: true` or by setting the environment variable `STUMP_ENABLE_UPLOAD = true`.
+
+When uploading is not enabled, the routes for uploading will not be attached to your server, and the web UI will not show the upload interface. Attempting to access uploading directly through the API will result in HTTP Error 404.
+
+### Permissions
+
+Only users with permission to upload are allowed to add files to the server. Users will need the following permissions:
+
+- `file:upload`
+- `manage:library`
+
+The server owner can assign these permissions to users on the user management settings page. For more information on that process, see the [user permissions](/guides/access-control/permissions) guide.
+
+### Configuration
+
+The server limits the maximum upload size based on your configuration, you can change `max_file_upload_size` in `Stump.toml` or the `STUMP_MAX_FILE_UPLOAD_SIZE` environment variable to set the maximum allowed upload size in bytes.
+
+## How it works
+
+This feature allows users to upload books or an entire collection of files and directories via series upload.
+
+After an upload finishes, a library scan will automatically be started to add the newly uploaded books.
+
+### Uploading books
+
+Users can upload one or more book files (e.g., `.cbz`, `.cbr`, `.epub`, and `.pdf` files) to a specific library directory.
+
+### Uploading Series
+
+Users can upload a `.zip` file containing multiple books in an arbitrary directory structure. The archive will be extracted to the location specified when uploading.
```

Comment:
I think it would be nice to use the stepped instruction pattern from a few other places, like the [scanner docs](https://www.stumpapp.dev/guides/basics/scanner#how-does-it-work). I think your title is much better though (i.e. not making the heading a title)

Reply:
Check out what I added. I'm pretty into the `InlineIcon` component I made (although if there was a pre-existing solution I missed let me know).

I added some steps, although in a slightly different way. Rather than walking through the steps that happen when an upload is done, I walked through the steps a user should take to make an upload happen. I thought that would be a good way to explain each of the upload request fields and what they do from a functional perspective.

Reply:
> Rather than walking through the steps that happen when an upload is done, I walked through the steps a user should take to make an upload happen

Nice, I think that’s perfect! I’ll also be able to take a look at the component you mentioned tomorrow